homework/index.js

```diff
@@ -30,15 +32,73 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
+  function renderContributors(contributors, contributorContainer) {
+    contributors.forEach(contributor => {
+      const contributorDetail = createAndAppend('div', contributorContainer, {
+        class: 'contributor-detail',
+      });
+      createAndAppend('img', contributorDetail, {
+        src: contributor.avatar_url,
+        alt: contributor.login,
+        class: 'avatar',
+      });
+      createAndAppend('a', contributorDetail, {
+        text: contributor.login,
+        href: contributor.html_url,
+        target: '_blank',
+      });
+      createAndAppend('p', contributorDetail, { text: contributor.contributions });
+    });
+  }
+
+  function dropDown(root, repos) {
+    const header = createAndAppend('header', root, { class: 'header' });
+    createAndAppend('h3', header, { text: 'HYF Repositories' });
+
+    const select = createAndAppend('select', header);
+    repos.forEach((repo, index) => {
+      createAndAppend('option', select, {
+        text: repo.name,
+        value: index,
+      });
     });
+
+    const container = createAndAppend('div', root, { id: 'container' });
+    const repoContainer = createAndAppend('div', container, { id: 'repo-container' });
+    const contributorContainer = createAndAppend('div', container, { id: 'contributor-container' });
+
+    select.addEventListener('change', () => {
+      const selectedRepo = repos[select.value];
+      console.log('name', selectedRepo);
+      repoContainer.innerHTML = '';
+      contributorContainer.innerHTML = '';
+
+      const updatedAt = new Date(selectedRepo.updated_at);
+
+      createAndAppend('p', repoContainer, { text: `Repository: ${selectedRepo.name}` });
```

Comment:
I wonder why you moved this logic in the event listener. In week1's homework, you had separate function for this logic which was better than doing it like this. Could you please explain the reasoning for doing it this way?